apps/anoma_node/lib/examples/e_shard_backend.ex

```diff
@@ -0,0 +1,718 @@
+defmodule Anoma.Node.Examples.EShardBackend do
+  @moduledoc """
+  I contain examples demonstrating transaction execution using the Shard backend.
+  """
+
+  alias Anoma.Node.Examples.ENode
+  alias Examples.ENock
+  alias Anoma.Node.Transaction.Mempool
+  alias Anoma.Node.Transaction.Shard
+  alias Anoma.Node.Registry
+  alias Anoma.Node.Event
+
+  require Noun
+  require Logger
+
+  import ExUnit.Assertions
+
+  @dialyzer :no_improper_lists
+
+  # --- Helper Functions ---
+
+  # Helper to wait for BlockEvent
+  defp wait_for_block(node_id, expected_round) do
+    block_filter = [
+      Event.node_filter(node_id),
+      %Mempool.BlockFilter{}
+    ]
+
+    EventBroker.subscribe_me(block_filter)
+
+    assert_receive(
+      %EventBroker.Event{
+        body: %Event{
+          node_id: ^node_id,
+          body: %Mempool.BlockEvent{round: ^expected_round}
+        }
+      },
+      # Timeout after 5 seconds
+      5000
+    )
+
+    EventBroker.unsubscribe_me(block_filter)
+    :ok
+  end
+
+  # Helper to wait for a specific condition based on a process's state
+  defp wait_for_state_condition(
+         pid,
+         description,
+         predicate_fun,
+         timeout \\ 5000
+       ) do
+    unless is_pid(pid) do
+      raise "Invalid PID provided to wait_for_state_condition: #{inspect(pid)}"
+    end
+
+    start_time = System.monotonic_time(:millisecond)
+    # ms
+    check_interval = 10
+
+    loop_check = fn fun ->
+      current_time = System.monotonic_time(:millisecond)
+
+      if current_time - start_time > timeout do
+        {:error, :timeout}
+      else
+        state = :sys.get_state(pid)
+
+        if predicate_fun.(state) do
+          :ok
+        else
+          Process.sleep(check_interval)
+          # Recurse
+          fun.(fun)
+        end
+      end
+    end
```

Comment:
We should not use `:sys.get_state`. This should be used very specifically and this isn't a valid use case.

Further can we remove the `Process.sleep` if you are waiting for something, why not subscribe to an event that you emit to do this. We should not sleep here